### PR TITLE
fix: job queue test failures

### PR DESCRIPTION
**Pull request by AmanuelAaron**
## Description

Fixing generic task tests that cause the job queue test to fail

## Test Plan

e2e tests should pass, specifically `test_job_queue.py` & `test_generic_tasks.py`

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: e2e_tests/tests/task/test_generic_tasks.py [hunk 1]
```diff
@@ -26,7 +26,17 @@ def test_create_generic_task() -> None:
...
+    res = subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+    id_index = res.stdout.find("Created task ")
+    task_id = res.stdout[id_index + len("Created task ") :].strip()
+
+    test_session = api_utils.determined_test_session()
+    is_valid_state = task.wait_for_task_state(
+        test_session, task_id, bindings.v1GenericTaskState.COMPLETED, timeout=30
+    )
+    if not is_valid_state:
```
Comment (ioga) [inline e2e_tests/tests/task/test_generic_tasks.py:38]: I recall `wait_for_task_state` used to do `pytest.fail`, same as `wait_for_experiment_state`. now it returns bool and you end up having this annoying check  in every test. 
do we really need an utility which returns bool? maybe we could have two helpers, one which `pytest.fails`, and another which returns the bool or the unexpected final state?

Comment (AmanuelAaron) [reply e2e_tests/tests/task/test_generic_tasks.py]: Updated the helper, as well as the other `wait_task_start`, to do `pytest.fails`

File: master/pkg/tasks/task_generic.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ func (s GenericTaskSpec) SetJobPriority(priority int) error { return nil }
 
 // SetWeight todo.
 func (s GenericTaskSpec) SetWeight(weight float64) error {
-	s.Base.ResourcesConfig.SetWeight(weight)
+	s.GenericTaskConfig.Resources.SetWeight(weight)
```
Comment (NicholasBlaskey) [inline master/pkg/tasks/task_generic.go:102]: will this actually affect priority in scheduling?

I see commands set the group weight in the resource manager

https://github.com/determined-ai/determined/blob/ba03375270d0c63620c4171a35a40d89ff4bf89a/master/internal/command/command_job_service.go#L62-L72

File: e2e_tests/tests/task/test_generic_tasks.py [hunk 3]
```diff
@@ -168,6 +176,13 @@ def test_generic_task_create_with_fork() -> None:
     expected_config = {"entrypoint": ["echo", "forked"]}
     assert result_config == expected_config
 
+    task.wait_for_task_state(
+        test_session, task_resp.taskId, bindings.v1GenericTaskState.COMPLETED, timeout=30
```
Comment (NicholasBlaskey) [inline e2e_tests/tests/task/test_generic_tasks.py:180]: I prefer if ```timeout=30``` is just a default value for the parameter in the function

Then we can override it if there are specific cases where we should only wait a time period. Most of these seem like a "just wait long enough"

File: master/internal/api_generic_tasks.go [hunk 4]
```diff
@@ -148,7 +148,7 @@ func (a *apiServer) getGenericTaskLaunchParameters(
 	taskSpec.UserSessionToken = token
 
 	genericTaskSpec.Base = taskSpec
-	genericTaskSpec.GenericTaskConfig = taskConfig
+	genericTaskSpec.GenericTaskConfig = &taskConfig
```
Comment (NicholasBlaskey) [inline master/internal/api_generic_tasks.go:151]: this is a metacomment but do you wanna make two PRs here?

One for the test failure (we think it is caused by not waiting for some tasks)

And a second PR (with tests) for the job queue not being restored and priority set?

File: master/pkg/tasks/task_generic.go (hunk 2 above)
Comment (AmanuelAaron) [reply master/pkg/tasks/task_generic.go:102]: The implementation of `toV1Job` and `SetWeight` was originally to just allow `det job list` to work when a generic task was running. We will still need to fully implement all the functions later. If you see the priority and resource pool functions are still unimplemented

File: master/internal/api_generic_tasks.go (hunk 4 above)
Comment (AmanuelAaron) [reply master/internal/api_generic_tasks.go]: Removed the job queue stuff from PR. Everything in this PR should make the tests pass

**Review by NicholasBlaskey — APPROVED** [review]

**Pull request merged**